context/AuthContext.tsx

```diff
@@ -0,0 +1,111 @@
+import {createContext, useContext, useEffect, useState} from "react"
+import {onAuthStateChanged} from 'firebase/auth'
+import {auth} from '../firebase/FirebaseConfig'
+import {fetchSignInMethodsForEmail,sendSignInLinkToEmail,isSignInWithEmailLink,signInWithEmailLink,signInWithPopup,signOut, GoogleAuthProvider,FacebookAuthProvider,GithubAuthProvider } from "firebase/auth";
+import {useAlert} from './AlertContext'
+import { OAuthProvider } from "firebase/auth";
+const AuthContext = createContext<any>({})
+
+export const useAuth =  () => useContext(AuthContext)
+
+// eslint-disable-next-line no-undef
+export const AuthContextProvider = ({children} : {children:React.ReactNode}) =>{
```

Comment:
see comment above / other comment regarding additional import to use `React.ReactNode`

Reply:
resolved in additional commit FYI